crates/router/src/connector/bluesnap.rs

```diff
@@ -760,3 +982,31 @@ impl api::IncomingWebhook for Bluesnap {
         Ok(res_json)
     }
 }
+
+impl services::ConnectorRedirectResponse for Bluesnap {
+    fn get_flow_type(
+        &self,
+        _query_params: &str,
+        json_payload: Option<serde_json::Value>,
+        _action: services::PaymentAction,
+    ) -> CustomResult<payments::CallConnectorAction, errors::ConnectorError> {
+        let redirection_response: bluesnap::BluesnapRedirectionResponse = json_payload
+            .ok_or(errors::ConnectorError::MissingRequiredField {
```

Comment:
add a new error for `MissingPayloadField{fieldname}`, MissingRequiredField is reserved for customer missing an required field. this is not customer input.